### PR TITLE
nixos/rustdesk-server: suppport enable rustdesk's signal server and relay server seperately and rename an option

### PR DESCRIPTION
**Pull request by wrvsrx**
In theory, we can use one server as rustdesk's signal server and another one as rustdesk's relay server. This PR allows such behavior.

This PR also renames `relayIP` to `relayHosts` since we can specify dns name or ip as a relay server.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wrvsrx** [comment]
@ofborg eval

File: nixos/modules/services/monitoring/rustdesk-server.nix [hunk 1]
```diff
@@ -18,30 +18,54 @@ in {
       '';
     };
 
-    relayIP = mkOption {
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/rustdesk-server.nix:21]: We should add mkRenamedOption for the options which got removed and can be mapped to new ones and mkRemovedOption for those who cannot be mapped.

File: nixos/modules/services/monitoring/rustdesk-server.nix [hunk 2]
```diff
@@ -18,30 +18,54 @@ in {
...
-      type = str;
-      description = ''
-        The public facing IP of the RustDesk relay.
-      '';
-    };
+    signal = {
+      enable = mkOption {
+        type = bool;
+        default = config.services.rustdesk-server.enable;
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/rustdesk-server.nix:24]:
```suggestion
        default = true;
```

File: nixos/modules/services/monitoring/rustdesk-server.nix [hunk 3]
```diff
@@ -18,30 +18,54 @@ in {
...
+        defaultText = literalExpression "config.services.rustdesk-server.enable";
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/rustdesk-server.nix:25]:
```suggestion
```

File: nixos/modules/services/monitoring/rustdesk-server.nix [hunk 4]
```diff
@@ -18,30 +18,54 @@ in {
...
+        description = ''
+          Whether to enable the RustDesk signal server.
+        '';
+      };
+
+      relayHosts = mkOption {
+        type = listOf str;
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/rustdesk-server.nix:32]: This is missing a default

File: nixos/modules/services/monitoring/rustdesk-server.nix [hunk 5]
```diff
@@ -18,30 +18,54 @@ in {
...
-      default = [];
-      example = [ "-k" "_" ];
-      description = ''
-        A list of extra command line arguments to pass to the `hbbr` process.
-      '';
+    relay = {
+      enable = mkOption {
+        type = bool;
+        default = config.services.rustdesk-server.enable;
+        defaultText = literalExpression "config.services.rustdesk-server.enable";
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/rustdesk-server.nix:54]:
```suggestion
        default = true;
```

**Pull request merged**